scripts/generate_artifacts.jl

```diff
@@ -0,0 +1,66 @@
+
+using ACE1pack, Pkg
+using Pkg.Artifacts
+using SHA
+
+artifacts_toml = joinpath(pathof(ACE1pack)[1:end-16], "Artifacts.toml")
+
+function _get_sha256(filename)
```

Comment:
Is this sound? Will `return` close the file? 

Alternatively, I thought about 
```
function _get_sha256(filename)
    open(filename) do f
        global hash = bytes2hex(sha256(f))
    end
    return hash
end
```
but it seemed wrong to make `hash` accessible across the script. 

It wasn't clear to me what is the correct way to access anything derived from the file contents outside of the `do` block. Maybe in that case one should explicitly `close()` the file?

Reply:
I'm afraid I have no memory about how this works. Have you checked what I did in `ACE1.jl`?

Reply:
Artifacts.toml: 
```
["acedata_v0.8.1"]
git-tree-sha1 = "23a2e3400188336105a4e47c31cdb7ce95a104cb"
lazy = true

    [["acedata_v0.8.1".download]]
    sha256 = "91305df8a0b0572630b9be4b2f3b7e16fc14b2683182e25f18a338036eac0305"
    url = "https://github.com/ACEsuit/ACEData/archive/v0.8.1.tar.gz"
```

and then in `artifacts.jl`: 
```Julia
using Pkg.Artifacts

datadir = joinpath(artifact"acedata_v0.8.1", "ACEData-0.8.1")
testsdir = joinpath(datadir, "tests")
traindir = joinpath(datadir, "trainingsets")
potsdir = joinpath(datadir, "potentials")

# unzip the eam potential
if !isfile(joinpath(traindir, "w_eam4.fs"))
   eamzip = joinpath(traindir, "w_eam4.fs.zip")
   run(`unzip $eamzip -d $traindir`)
end
```

Reply:
why do you even need the sha in `artifacts.jl`?

Reply:
> why do you even need the sha in artifacts.jl?

`bind_artifact!()` needs both sha and url and you need the url to get the artefact. 

> I thought the hash can be just anything (compatible with whatever keys they use...) but you are saying this is not the case?

hashes for the downloaded file and file used to create the artefact need to match, no? I thought the initial error was raised because they didn't. 

ACE1 artefacts are successfully downloaded with what I thought was equivalent code, so maybe the issue is with how the original artefact files were set up. I'll see if I can make any progress tomorrow